---
title: 'Alveolar Macrophages after Murine Lung Transplant '
author: '180166459'
date: "2/8/2021"
output:
  word_document: default
---
```{r echo = FALSE, eval = FALSE}
version #check version of RStudio - this is relevant for packages 
```

```{r setup, include=FALSE}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r echo = FALSE, eval = FALSE}
#Clear Workspace
rm(list = ls(all.names = TRUE)) # clear all objects includes hidden objects.
gc() #frees up memory and reports the memory usage
```

## R Markdown

# **BMS353**

# **Bioinformatics**

**Alveolar Macrophages after Murine Lung Transplant -
(mouse)**

FOR NOT INCLUDING R CODE IN REPORT: {r echo = FALSE}
FOR NOT INCLUDING OUTPUT/CONSOLE PRINT in REPORT: {r eval = FALSE}

This is an R Markdown document. Markdown is a simple formatting syntax.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:


**INTRODUCTION**

! [Alt text] (Users\Andr\Documents\BMS353-Alveolar-Macrophages---RNA-seq-dataset\images\image1macrophages.jpg)

In this report we will analyze data for a research study that covers the following topic: Alveolar Macrophages after Murine Lung Transplant. For this we have the following resources provided:

*METHODS*


|Sample no. | Condition |
|-------:|:------|
| 4   | naive|
| 4 | 2h post-perfusion |
| 4 | 24 h post-perfusion |

[GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE116583)

[Google drive](https://drive.google.com/drive/folders/1AqYi0Ps5t5xo6XYWXjOtO6eDOMpT1ADE) - Google drive folder with the following content:

* meta_data
* salmon-quant.sf files
* tx2gene.csv


The project name BMS353-Alveolar-Macrophages---RNA-seq-dataset with all files and the code are available on github in a repository:

[Github](https://github.com/IoanaAndra/BMS353-Alveolar-Macrophages---RNA-seq-dataset)


In order to use the necessary functions and analyze the data, we use some packages available in R:

* readr
* ggplot2
* dplyr

Every chunk of code can be name like this:
```{r Name_of_chunk_here}

```

Installing packages in R

```{r eval = FALSE, tidy =FALSE}
#SECTION FOR PACKAGES
#installing all necessary packages
# IF using same R version (4.0.3) and have all packages installed already, all lines of this section can remain commented 

 
# install.packages("readr")
# install.packages("ggplot2")
# install.packages("dplyr")
# install.packages("learnr")
# install.packages("stringi")
# install.packages("tidyverse")
install.packages("ggpubr")
# if (!requireNamespace("BiocManager", quietly = TRUE))     # installing tximport
#     install.packages("BiocManager")
# 
# BiocManager::install("tximport")

# install.packages("jsonlite")
# BiocManager::install("GenomicFeatures")

install.packages("S4Vectors")
install.packages("stats4")
install.packages("BiocGenerics")



install.packages() #check installed packages. This can be seen also in Packages tab

# old.packages() 

# update.packages(ask = FALSE) #update installed packages without asking permission from user

```

CSV and TSV file formats can be stored in variables and display first row of them using head()

With head, we display first rows of data tx2gene.csv and a quant.sf

tx2gene.csv is the transcripts and genes relationship table file
```{r Read tx2gene.csv, tidy =FALSE, collapse = FALSE} 
library(readr)

dataPath <- "tx2gene.csv" #asigning value which represent path of file

file.exists(dataPath) #file.exists() return TRUE id file can be found or FALSE if it is not found
tx2GeneFile <- read.csv(file = 'tx2gene.csv') #file with delimiter ","
quantFile <- read_tsv(file = 'salmon_quant/SRR7457551/quant.sf') #file with delimiter "\t"
head(tx2GeneFile)
head(quantFile)
#view(tx2GeneFile)
```
Name represents in quant. sf the TXNAME found in tx2gene.csv. But tx2gene.csv contains all 12 samples (in order ...51-...62)


**ANALYSIS**

Sample txt. presents the full experimental design that was followed. 12 columns for all 12 samples that are found in a salmon output format file that can be further read and accessed using read_tsv
```{r tidy = FALSE, collapse = FALSE}


library(tximport)
library(readr)
library(dplyr)
# Read the sample information into R
sampleinfo <- read.delim("meta_data/sampleInfo.txt") #reads in delimited text file
View(sampleinfo)    # view sampleinfo.txt in a new tab
#glimpse(sampleinfo)
sampleinfo


rownames(sampleinfo) <- sampleinfo$run 

dirs <- list.files("salmon_quant/")
quant_files <- list.files("salmon_quant/",pattern="quant.sf",recursive = TRUE,full.names = TRUE)
#names(quant_files) <- dirs
quant_files

# tx2gene links transcript IDs to gene IDs for summarization
tx2geneVariable <- read.csv("tx2gene.csv")

txi <- tximport(files=quant_files, type="salmon", tx2gene=tx2geneVariable)

View(txi)  #view txi in a new tab

#inspect salmon output (quant.sf files)
quants <- read_tsv(quant_files[1])
head(quants)

 # spec(quants)
 
 filter(quants, quants$TPM ==0) #filter data where TPM = 0
#print.data.frame(quants) #print quants, maximum output to print in console 200 lines
 
tail(quants, n = 3)

max(quants$TPM)


 
```


After asigning to quants variable the files


```{r tidy = FALSE, collapse = FALSE}
rpk <- quants$NumReads / quants$EffectiveLength
scale_factor <- sum(rpk) / 1e6
tpm <- rpk / scale_factor

#define transcript mapping

gtf_file <- "Mus_musculus.GRCm38.91.chr.gtf.gz"
file.exists(gtf_file)

download.file("ftp://ftp.ensembl.org/pub/release-91/gtf/mus_musculus/Mus_musculus.GRCm38.91.chr.gtf.gz",destfile = gtf_file) #gtf based on organism of interest

# create a database of transcripts
# Could take a few minutes to run the makeTxDbFromGFF command
library(GenomicFeatures)
txdb <- makeTxDbFromGFF(gtf_file)


#specify number of keys and columns

keytypes(txdb)
columns(txdb)

#get names for all transcripts - using keys function
#compose query - using select function - this will return data frame

k <- keys(txdb, keytype="TXNAME")
tx_map <- select(txdb, keys = k, columns="GENEID", keytype = "TXNAME")

#visualise first rows of the transcript map
head(tx_map)
```



``` {r eval = FALSE, tidy = FALSE}
#use tximport package

library(tximport)
tx2gene <- tx_map
# write.csv(tx2gene,file="tx2gene.csv",row.names = FALSE,quote=FALSE)
txi <- tximport(quant_files,type="salmon",tx2gene = tx2gene)

table(tx_map$TXNAME %in% quants$Name)

tx2gene <- tx_map
txi <- tximport(quant_files,type="salmon",tx2gene = tx2gene,ignoreTxVersion = TRUE)

names(txi)

head(txi$counts) 

all(rownames(sampleinfo) == colnames(txi$counts))

library(tidyr)
library(dplyr)

quants <- separate(quants, Name, c("TXNAME","Number"),remove = FALSE)
head(quants)


quants <- left_join(quants, tx_map, by="TXNAME")
head(quants)

tx2gene <- dplyr:::select(quants, Name, GENEID)
head(tx2gene)

any(is.na(tx2gene$GENEID))

tx2gene <- filter(tx2gene, !is.na(GENEID))

library(tximport)
txi <- tximport(quant_files,type="salmon",tx2gene = tx2gene)



```
Summarising
```{r}
library(readr)
library(dplyr)

head(quants)
summarise(quants, min(quants$Length), max(quants$Length), min(quants$EffectiveLength),max(quants$EffectiveLength)) #summarises in a table values that we select from object 

write.table(colnames(quants), col.names=FALSE)
```
Next, testing quality assesment for the raw reads of gene transcripts that were imported with tximport.
```{r}
library(DESeq2)

dds <- DESeqDataSetFromTximport(txi, 
                                colData = sampleinfo,
                                design <- ~condition)


colData(dds)

tpm <- txi$abundance #asign list of TPM values to tpm variable
write.csv(tpm, file="tpm_values_abundance.csv",quote=FALSE)

```

Saving in CSV. Can be used in GOrilla
```{r}
library(readr)
library(dplyr)

quantFile <- read_tsv(file = 'salmon_quant/SRR7457551/quant.sf')
write.csv(quantFile, sep=",", file = 'Salmon outputs converted into CSV/SRR7457551.csv')

tx2gene <- dplyr:::select(quants, Name, GENEID)
head(tx2gene)

any(is.na(tx2gene$GENEID))

tx2gene <- filter(tx2gene, !is.na(GENEID))
# write.table(quantFile, file="", quote=TRUE, sep=",", row.names=TRUE)
```



```{r}
library(readr)
library(ggplot2)
library(data.table)

# head(quants)

quantFile_51 <- read_tsv(file = 'salmon_quant/SRR7457551/quant.sf', show_col_types = FALSE)


plotVar <- ggplot(quantFile_51, aes(x = quantFile_51$TPM, y = quantFile_51$Length)) +
  geom_point()
plotVar + theme_bw() + scale_x_continuous(trans='log10')

# plotVar + theme_minimal()

# mean(quants$Length) mean(quants$EffectiveLength)


text <- paste("text for this plot", "newline", sep = " ")

text.p <- ggparagraph(text = text, face = "italic", size = 11, color = "black")
```
```{r}
library(readr)
library(ggplot2)
library(data.table)
library(ggpubr)

quantFile_53 <- read_tsv(file = 'salmon_quant/SRR7457553/quant.sf', show_col_types = FALSE)


plotVar2 <- ggplot(quantFile_53, aes(x = quantFile_53$TPM, y = quantFile_53$Length)) +
  geom_point()

# plotvar2 + theme_bw() + scale_x_continuous(trans='log10')     #Error: object 'plotvar2' not found
# 
# #adding legend

text <- paste("text for this plot", "newline", sep = " ")

text.p <- ggparagraph(text = text, face = "italic", size = 11, color = "black")
```




```{r}

library(ggplot2)
library(data.table)


quantFile_53 <- read_tsv(file = 'salmon_quant/SRR7457553/quant.sf') #naive sample

quantFile_51 <- read_tsv(file = 'salmon_quant/SRR7457551/quant.sf') #2hpost-perfusion sample


plotVar <- ggplot(quantFile_51, aes(x = quantFile_51$TPM, y = quantFile_51$Length)) +
  geom_point() + ggplot(quantFile_53, aes(x = quantFile_53$TPM, y = quantFile_53$Length)) + geom_point()
plotVar + theme_bw()

plotVar + scale_x_continuous(trans='log10')


# geom_point() + geom_point(data, color = "red") # re-define data and overwrite top layer inheritance
```


```{r eval = FALSE, echo = FALSE}


rlang::last_error() # to check last error


```




library(readr)
library(dplyr)

quantFile <- read_tsv(file = 'salmon_quant/SRR7457551/quant.sf')



**CONCLUSIONS**





*References*
