---
title: "Alveolar Macrophages after Murine Lung Transplant "
author: "Ioana-Andra Popovici"
date: "2/8/2021"
output: word_document
---
```{r}
version #check version of RStudio - this is relevant for packages 
```

```{r setup, include=FALSE}
```


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


**BMS353**

**Bioinformatics**

**Alveolar Macrophages after Murine Lung Transplant -
(mouse)**


**INTRODUCTION**

In this report we will analyze data for a research study that covers the following topic: Alveolar Macrophages after Murine Lung Transplant. For this we have the following resources provided:

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE116583

https://drive.google.com/drive/folders/1AqYi0Ps5t5xo6XYWXjOtO6eDOMpT1ADE - Google drive folder with the following content:

* meta_data
* salmon-quant.sf files
* tx2gene.csv


The project name BMS353-Alveolar-Macrophages---RNA-seq-dataset with all files and the code are available on github in a repository:

https://github.com/IoanaAndra/BMS353-Alveolar-Macrophages---RNA-seq-dataset



In order to use the necessary functions and analyze the data, we use some packages available in R:

* readr
* ggplot2
* dplyr

```{r}
#SECTION FOR PACKAGES
#installing all necessary packages
# IF using same R version (4.0.3) and have all packages installed already, all lines of this section can be commented 


install.packages("readr")
install.packages("ggplot2")
install.packages("dplyr")
install.packages("learnr")




```
```{r}

library(readr) #a package used to import spreadsheets in R

dataset_path <- "tx2gene.csv" #asigning value which represent part of the path, in this case, just the name of the file worked

file.exists(dataset_path) #file.exists() return TRUE id file can be found or FALSE if it is not found


data_genes <- read_csv(dataset_path) #storing content of csv file into variable data_genes

#check data frame that we created

test_data <- read_table(data_genes) ##*******Error: `file` must be a string, raw vector or a connection.

View(test)  #view() - invoking a spreadsheet style data viewer nb


head(data_genes)

```



**ANALYSIS**


```{r}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("tximport")

library(tximport)
library(readr)
# Read the sample information into R
sampleinfo <- read.delim("meta_data/sampleInfo.txt")
View(sampleinfo)
sampleinfo


rownames(sampleinfo) <- sampleinfo$run




txi <- tximport(files=..., type="salmon", tx2gene=...)

dirs <- list.files("salmon_quant/")
quant_files <- list.files("salmon_quant/",pattern="quant.sf.gz",recursive = TRUE,full.names = TRUE)
names(quant_files) <- dirs
quant_files

#inspect salmon output (squant.sf files)


library(readr)
quants <- read_tsv(quant_files[1])


head(quants)
```


```{r}
rpk <- quants$NumReads / quants$EffectiveLength
scale_factor <- sum(rpk) / 1e6
tpm <- rpk / scale_factor


#define transcript mapping

gtf_file <- "Mus_musculus.GRCm38.91.chr.gtf.gz"
file.exists(gtf_file)


download.file("ftp://ftp.ensembl.org/pub/release-91/gtf/mus_musculus/Mus_musculus.GRCm38.91.chr.gtf.gz",destfile = gtf_file) # you have to change the link. Navigate and find the organism of interest

# create a database of transcripts
## Could take a few minutes to run the makeTxDbFromGFF command
library(GenomicFeatures)
txdb <- makeTxDbFromGFF(gtf_file)


#specify number of keys and columns

keytypes(txdb)
columns(txdb)

#get names for all transcripts - using keys function
#compose query - using select function - this will return data frame

k <- keys(txdb, keytype="TXNAME")
tx_map <- select(txdb, keys = k, columns="GENEID", keytype = "TXNAME")

#visualise first rows of the transcript map
head(tx_map)


#use tximport package

library(tximport)
tx2gene <- tx_map
write.csv(tx2gene,file="tx2gene.csv",row.names = FALSE,quote=FALSE)
txi <- tximport(quant_files,type="salmon",tx2gene = tx2gene)

table(tx_map$TXNAME %in% quants$Name)


library(tximport)
tx2gene <- tx_map
txi <- tximport(quant_files,type="salmon",tx2gene = tx2gene,ignoreTxVersion = TRUE)


```



**CONCLUSIONS**





*References*
